src/file/reader.rs

```diff
@@ -285,21 +285,45 @@ impl PageReader for SerializedPageReader {
   fn get_next_page(&mut self) -> Result<Option<Page>> {
     while self.seen_num_values < self.total_num_values {
       let page_header = self.read_page_header()?;
-      let compressed_len = page_header.compressed_page_size as usize;
-      let uncompressed_len = page_header.uncompressed_page_size as usize;
-      let mut buffer = vec![0; compressed_len];
+
+      // when processing data page v2, depending on enabled compression for the page, we should
+      // account for uncompressed data ('offset') of repetition and definition levels.
+      let (offset, can_apply_compression) = if page_header.data_page_header_v2.is_some() {
```

Comment:
nit: this can be replaced by `if let ...`. Also, maybe change `can_apply_compression` to something like `is_compressed` or `can_decompress`?

Reply:
Good point. Will change and give variable a more meaningful name.